scripts/cbmc_json_parser.py

```diff
@@ -165,6 +171,60 @@ def extract_solver_information(cbmc_response_json_array):
 
     return properties, solver_information
 
+def postprocess_results(properties):
+    """
+    Check for certain cases, e.g. a reachable unsupported construct or a failed
+    unwinding assertion, and update the results of impacted checks accordingly.
+    1. Change all "SUCCESS"/"FAILURE" results to "UNKNOWN" if the reachability
+    check for a Rust construct that is not currently supported by RMC failed,
+    since the missing modeling of the unsupported construct may impact results.
+    2. TODO: Change results from "SUCCESS" to "UNKNOWN" if an unwinding
+    assertion failed, since the insufficient unwinding may cause some execution
+    paths to be left unexplored.
+    Additionally, print a message at the end of the output that indicates if any
+    of the special cases above was hit.
+    """
+
+    has_reachable_unsupported_constructs, has_failed_unwinding_asserts = check_special_cases(properties)
+
+    for property in properties:
+        if has_reachable_unsupported_constructs:
+            # Change status of all properties except failed unsupported
+            # construct reachability checks to UNKNOWN
+            description = property["description"]
+            if GlobalMessages.UNSUPPORTED_CONSTRUCT_DESC not in description or property["status"] == "SUCCESS":
+                property["status"] = "UNKNOWN"
+        # TODO: Handle unwinding assertion failure
+
+    messages = ""
+    if has_reachable_unsupported_constructs:
+        messages += "** WARNING: A Rust construct that is not currently supported " \
+                    "by RMC was found to be reachable. Check the results for " \
+                    "more details."
+
+    return properties, messages
+
+
+def check_special_cases(properties):
```

Comment:
nit: I would change this to be:

```python
def has_check_failure(properties, message):
    for property in properties:
        if message in property["description"] and property["status"] == "FAILURE":
                return True
    return False
```
Then change the caller to invoke this twice. I think it will make this easier to read.

Reply:
I was trying to avoid iterating over all properties (which could be in the tens of thousands for larger tests) multiple times. I could refactor the loop so that it iterates over checks passed by the caller, e.g.
```python
def check_special_cases(properties, cases):

    for property in properties:
        for case in cases:
            if case.description in property["description"] and property["status"] == case.result:
            ...
```
but, I'm not sure if it's worth it given that we're likely getting rid of this script soon. Let me know your thoughts.

Reply:
The current code is iterating over all properties though, isn't it? But I don't think it matters. It should still be pretty fast and you can't avoid it in successful cases anyway.